scss/_patterns_buttons.scss

```diff
@@ -80,7 +80,7 @@
       $button-disabled-border-color: $colors--theme--button-positive-default,
       $button-text-color: $colors--theme--button-positive-text
     );
-    @include vf-focus($color-focus-positive);
+    @include vf-focus($colors--theme--positive-focus);
```

Comment:
Would be great if we don't have to introduce a new theme colour just for the sake of one variant in single component. Seems like a bit overkill.

We don't have any good means of components to "add" colours to themes (themes are defined centrally), but I wonder if we could utilise this somehow: 
https://github.com/canonical/vanilla-framework/blob/4c15e4009ba37baa3c6b7ef4a6b3f386d766a461/scss/_base_themes.scss#L47-L52

It uses a CSS variable with theme flag (light/dark) to switch between 2 values.

Reply:
> An alternative might be adding another helper that just returns a value based on theme (that could then be passed to functions like `vf-focus`), but I think that's a bit out of scope for this PR.

Agree, this is the best way to do this but we needn't implement it here. 

Maybe we should implement this with a new variable for now, and clean things up later?

Reply:
It's fine for now, but I kind of don't like the precedent of us fixing this by adding new theme variable for a single variant of a single component. We probably wouldn't want to add a new design token for that, but that's kind of what we are doing.

Reply:
> Unfortunately the use of `vf-focus` for focus states kind of precludes using helpers like this

@pastelcyborg Of course it would not with current `vf-focus`. This focus placeholder is an old piece of code that was clearly not created with theming in mind. At first when migrating to new theming I assumed we will still use the same focus color regardless of the theme, so this was not addressed in any way.

But it looks like we need to address it. I created https://warthogs.atlassian.net/browse/WD-13363 to cover this.